.github/workflows/pypi-cd.yml

```diff
@@ -20,6 +20,23 @@ permissions:
       id-token: write
 
 jobs:
+    output-matrix:
+        runs-on: ubuntu-latest
+        outputs:
+            MATRIX: ${{ steps.output-matrix.outputs.MATRIX }}
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v4
+
+            - name: output-matrix
+              id: output-matrix
+              shell: bash
+              run: |
+                # Get the matrix from the matrix.json file, without the object that has the IMAGE key
```

Comment:
If we want to reuse the the matrix we should have some field in each entry to say which languages is supported,for example:
 ```
 {
        "OS": "macos",
        "NAMED_OS": "darwin",
        "RUNNER": "macos-13-xlarge",
        "ARCH": "arm64",
        "TARGET": "aarch64-apple-darwin",
        "PACKAGE_MANAGERS": ["pypi", "npm"]
    },
```
and then to filter it based on this PACKAGE_MANAGERS field